kcidb/cloud/run.sh

```diff
@@ -263,6 +265,8 @@ YAML_END
                     limits:
                       cpu: "0.25"
                       memory: "256M"
+                  args:
+                    - $(yaml_quote "$topic")
```

Comment:
Don't pass it as arguments, instead pass it as an environment variable (e.g. called `KCIDB_NEW_TOPIC`). See the Grafana deployment above, under the `env` key, or read the reference, although it's not very nice: https://cloud.google.com/run/docs/reference/yaml/v1

The project name is likely defined by default as `PROJECT_ID`.

You can use the `kcidb.mq.IOPublisher` class to publish the data: https://github.com/kernelci/kcidb/blob/ded2da11dd92a94466991f3f35b49e6ab048550f/kcidb/mq/__init__.py#L369

Something like this:
```python
client = kcidb.mq.IOPublisher(project_id, topic_name)
client.publish(issue_and_or_incident_data)
```

Reply:
@spbnick Thanks for all the help, Nick.
The above method for publishing the data did work.
Bdw just to understand, could you please explain a bit more about `IOPublisher` class and how it manages communications between different deployment instances?

Reply:
`IOPublisher` is a thin wrapper around the [Google Cloud Pub/Sub publisher client](https://cloud.google.com/python/docs/reference/pubsub/latest/google.cloud.pubsub_v1.publisher.client.Client). All it does is validate and encode messages, and then publish them to a topic. See an overview of Pub/Sub, if you'd like an intro: https://cloud.google.com/pubsub/docs/overview

The `IOPublisher` doesn't really "manage communication between deployment instances", all it does is publish stuff to a topic. It depends on which credentials (normally supplied via environment variables) it's using and permissions set up on the project and the topic in the Google Cloud, whether it would be able to publish to a different instance. It doesn't really know anything instances/deployments.

However, the idea of KCIDB deployments is that each is self-contained. A deployment does everything KCIDB does, it's just a copy of KCIDB setup. And so the IOPublisher we'll be creating and using, will be submitting data *within* its deployment. That's ensured by providing it with appropriate parameters within the `cloud` tool and the code you're writing. Once you're done implementing and testing this, and we merge your PR, the staging, playground, and production deployments will be updated to include your setup. And each deployment will get its own instance of the editor.

Reply:
Thanks for all the info Nick :) Much appreciated!